lua/starfall/libs_cl/bass.lua

```diff
@@ -31,12 +106,15 @@ SF.RegisterType("Bass", true, false)
 return function(instance)
 local checkpermission = instance.player ~= SF.Superuser and SF.Permissions.check or function() end
 
--- Register functions to be called when the chip is initialised and deinitialised
-local sounds = {}
+-- Register functions to be called when the chip is initialized and deinitialized.
+local soundList = {}
+
 instance:AddHook("deinitialize", function()
-	for s in pairs(sounds) do
-		deleteSound(instance.player, s)
+	for _, snd in ipairs(soundList) do
+		deleteSound(instance.player, snd)
 	end
+
+	soundList = nil
 end)
```

Comment:
Not sure why this was changed to an array and the name changed. Could those changes be reverted?

Reply:
This would also be causing a lua error on line 441.

Reply:
Previously, `sounds` was used to both track sounds per-instance so they can be deleted, and store extra data about them.
Now, `soundDatas` stores the extra data (and does so globally), so we only need to keep track of a list of sounds created by the instance; it's cleaner for it to not be a lookup table any more.